doc/api/n-api.md

```diff
@@ -3148,6 +3150,31 @@ Returns `napi_ok` if the API succeeded.
 This API represents the invocation of the Strict Equality algorithm as
 defined in [Section 7.2.14][] of the ECMAScript Language Specification.
 
+### napi_detach_arraybuffer
+<!-- YAML
+added: REPLACEME
+-->
+
+```C
+napi_status napi_detach_arraybuffer(napi_env env,
+                                    napi_value arraybuffer)
+```
+
+* `[in] env`: The environment that the API is invoked under.
+* `[in] arraybuffer`: The JavaScript ArrayBuffer to be detached.
+
+Returns `napi_ok` if the API succeeded. If a non-detachable ArrayBuffer is
+passed in it returns `napi_detachable_arraybuffer_expected`.
+
+Generally, an ArrayBuffer would be non-detachable if its been detached before.
+And another engine specific condition is that an ArrayBuffer would be detachable
+only if it is an external ArrayBuffer created by
+[`napi_create_external_arraybuffer`][]. V8 requires the ArrayBuffer being
+external to be detachable.
```

Comment:
I’m a bit torn on that – I think I *would* keep this restriction here *because* it’s supposed to be engine-agnostic. If other engines do not have this restriction, then we should attempt to emulate V8’s behaviour for that case everywhere.

Reply:
@addaleax OK, so how about

> An &#96;ArrayBuffer&#96; is non-detachable if it has been detached before. The engine may impose additional conditions on whether an &#96;ArrayBuffer&#96; is detachable. For example, V8 requires that the &#96;ArrayBuffer&#96; be external, that is, created with [&#96;napi_create_external_arraybuffer&#96;][].